cli/popper/commands/cmd_matrix.py

```diff
@@ -0,0 +1,58 @@
+import click
+import popper.utils as pu
+
+from popper.cli import pass_context
+from popper.exceptions import BadArgumentUsage
+
+
+@click.command('matrix', short_help='Define execution matrix for a '
+               'pipeline using environment variables.')
+@click.argument('pipeline', required=True)
+@click.option(
+    '--add',
+    multiple=True,
+    help='Add environment variable in the form key=value',
+    required=False
+)
+@click.option(
+    '--rm',
+    multiple=True,
+    help='Remove environment variable using the key',
+    required=False
+)
+@pass_context
+def cli(ctx, pipeline, add, rm):
+    try:
+        config = pu.read_config()
+        pipeline_config = config['pipelines'][pipeline]
+    except KeyError:
+        raise BadArgumentUsage("Pipeline {} does not exist.".format(pipeline))
+    if not add and not rm:
+        try:
+            pu.print_yaml(pipeline_config['matrix']['vars'])
+        except KeyError:
+            pu.info("No environment variables defined for this pipeline.")
+    if add:
+        try:
+            matrix = pipeline_config['matrix']
+        except KeyError:
+            matrix = {'vars': {}}
```

Comment:
since we always expect `vars`, maybe it should be added as part of `popper init`. Also, and more importantly, `vars` is a list of dictionaries. For every new item in the one-dimensional matrix, we add a dictionary, where the dictionary has as many key-value pairs. We can discuss this further on gitter if you want.